tests/src/test/scala/com/nvidia/spark/rapids/unit/StringUnitTest.scala

```diff
@@ -0,0 +1,159 @@
+                assertResult(hostResult.getJavaString(0))(expectedStr)
+              }
+          }
+        }
+
+      doTest(-1, "")
+      doTest(0, "")
+      doTest(1, "abc123")
+      doTest(2, "abc123abc123")
+      doTest(3, "abc123abc123abc123")
+    }
+  }
+
+  test("Test StringRepeat with scalar string and column repeatTimes") {
+    val intCol = ColumnVector.fromBoxedInts(-3, null, -1, 0, 1, 2, null)
+
+    withResource(GpuColumnVector.from(intCol, DataTypes.IntegerType)) { intGpuColumn =>
+      // Test repeat(NULL, intVal).
+      withResource(GpuScalar(null, DataTypes.StringType)) { nullStr =>
+        withResource(GpuStringRepeat(null, null).doColumnar(nullStr, intGpuColumn)) { result =>
+          withResource(result.copyToHost()) { hostResult =>
+            (0 until hostResult.getRowCount.asInstanceOf[Int]).foreach { i =>
+              assertResult(hostResult.isNull(0))(true)
+            }
+          }
+        }
+      }
+
+      // Test repeat(str, intVal).
+      withResource(GpuScalar("abc123", DataTypes.StringType)) { strScalar =>
+        withResource(GpuStringRepeat(null, null).doColumnar(strScalar, intGpuColumn)) { result =>
+          withResource(result.copyToHost()) { hostResult =>
+            CudfTestHelper.assertColumnsAreEqual(
+              HostColumnVector.fromStrings("", null, "", "", "abc123", "abc123abc123", null),
+              hostResult,
+              "Repeated strings")
+          }
+        }
+      }
+    }
+  }
+
+  test("Test StringRepeat with column strings and scalar repeatTimes") {
+    val strsCol = ColumnVector.fromStrings(null, "a", "", "123", "éá")
+
+    withResource(GpuColumnVector.from(strsCol, DataTypes.StringType)) { strGpuColumn =>
+      // Test repeat(strs, NULL).
+      withResource(GpuScalar(null, DataTypes.IntegerType)) { nullInt =>
+        withResource(GpuStringRepeat(null, null).doColumnar(strGpuColumn, nullInt)) { result =>
+          withResource(result.copyToHost()) { hostResult =>
+            (0 until hostResult.getRowCount.asInstanceOf[Int]).foreach { i =>
+              assertResult(hostResult.isNull(0))(true)
+            }
+          }
+        }
+      }
+
+      // Test repeat(strs, intVal).
+      val doTest = (intVal: Int, expectedStrs : HostColumnVector) =>
+        withResource(GpuScalar(intVal, DataTypes.IntegerType)) { intScalar =>
+          withResource(GpuStringRepeat(null, null).doColumnar(strGpuColumn, intScalar)) { result =>
+            withResource(result.copyToHost()) { hostResult =>
+              CudfTestHelper.assertColumnsAreEqual(expectedStrs, hostResult, "Repeated strings")
+            }
+          }
+        }
+
+      doTest(-10, HostColumnVector.fromStrings(null, "", "", "", ""))
+      doTest(0, HostColumnVector.fromStrings(null, "", "", "", ""))
+      doTest(2, HostColumnVector.fromStrings(null, "aa", "", "123123", "éáéá"))
+    }
+  }
+
+  test("Test StringRepeat with column strings and column repeatTimes") {
```

Comment:
Isn't this test already covered by the PySpark tests?

Reply:
It has been covered in PySpark test. I want to do that again here to maximize coverage.

Reply:
Our CI testing already takes too long, why would we want redundant tests if they aren't adding any value?

Reply:
Ok then I'm going to remove the repeated tests :smiley: .